go/nbs/store.go

```diff
@@ -290,13 +304,24 @@ func (nbs *NomsBlockStore) UpdateRoot(current, last hash.Hash) bool {
 		nbs.mt = nil
 	}
 
-	actual, tableNames := nbs.mm.Update(nbs.tables.ToSpecs(), nbs.root, current, nil)
+	candidate := nbs.tables
+	var compactees chunkSources
+	if candidate.Size() > nbs.maxTables {
```

Comment:
I mean because it could turn out that a memory table that is compacting will end up having 0 chunks (all dups), in which case our compactingChunkSource table shouldn't have been counted.

IOW, it seems like this point should synchronize with any memtable compaction that's taking place before continuing.